tls/s2n_security_rules.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_result.h"
+
+typedef enum {
+    S2N_PERFECT_FORWARD_SECRECY = 0,
+    S2N_SECURITY_RULES_COUNT,
+} s2n_security_rule_id;
+
+#define TO_FLAG(rule) rule##_FLAG = (1 << (rule))
+typedef enum {
+    TO_FLAG(S2N_PERFECT_FORWARD_SECRECY),
+} s2n_security_flag;
+
+struct s2n_security_rule_result {
+    bool found_error;
+    bool write_output;
```

Comment:
Is this field needed if you just key'ed on the stuffer being initialized or not?

Reply:
I started out doing that, but it's a little messy. Either we assume the stuffer is always growable and check "output->growable", or we need a more complex check like "!s2n_stuffer_is_freed(output) || output->growable".

I like the flag because it explicitly indicates that we expect output. If the stuffer isn't configured properly but the flag is set, we'll fail. Otherwise, if the stuffer isn't configured properly, we just skip output.

Reply:
Fair enough